lib/ach/string_formatting_helper.rb

```diff
@@ -4,7 +4,7 @@ module StringFormattingHelper
     def self.stringify_with_same_day(f)
       return f.upcase if f.to_s.upcase.match(/^SD\d+$/)
 
-      Date.strptime(f, '%y%m%d')
+      f.strftime('%y%m%d')
     rescue
       f
```

Comment:
Thanks for reporting the issue. The only problem is that this `f` value may not be a `Date`, it really can be any 6 characters according to the ACH specs.

What I believe we should do is make sure it's always 6 characters (which is what I thought `strptime` would do, my bad..) something like:

```suggestion
      Date.parse(f).strftime('%y%m%d')
    rescue
      f.rjust(6, " ")
```

Reply:
Updated the comment because we might want to do parse only if it's not a date.

Reply:
I understand your goal but there are a few problems

- `f` can be `Time` or `DateTime`
- in case if `f` match to `/^SD\d+$/` (e.g. `SD12345`)

Also, from the definition of the `.field` method, I see that it supports validation, maybe this is a more appropriate place for it?

Reply:
The case for `SDhhmm` strings is handled on line 5

Reply:
The `company_descriptive_date` field of the batch header is optional and alphanumeric, it doesn't have to be parse-able, so I don't see how we could validate it as it really can be any 6 characters

Reply:
I believe this covers all the cases:

```suggestion
      f = Date.parse(f) unless f.respond_to?(:strftime)

      f.strftime('%y%m%d')
    rescue
      f.to_s.rjust(6, ' ')
    end
```

It will:
- Try to parse the date only if `f` does not already respond to `strftime`
- If it fails, pad the string with spaces up to 6 characters

What do you think @chubchenko?